### ISSUE TITLE
[Feature Request] Horizon Chart

### ISSUE THREAD

**Issue by cassiobatista** [pyviz/holoviews#3806]
Following a [suggestion I received on Bokeh Discourse page](https://discourse.bokeh.org/t/old-bokeh-charts-horizon-no-longer-supported/3703), wouldn't it be nice to have a Horizon chart / plot in Holoviews similar to those Bokeh used to support in [earlier 0.12 versions](https://bokeh.pydata.org/en/0.12.4/docs/reference/charts.html#horizon)? 

![horizon](https://user-images.githubusercontent.com/2287025/60756033-bcbb9480-9fce-11e9-83f2-30a89d167af9.png)

**Comment by poplarShift** [comment]
I may be ignorant but I've never come across these. Quoting https://rpubs.com/ramanraja/horizon:

> (1) Both positive and negative values are plotted above the x axis. Color is used to differentiate between them. Positives are blue, negatives are red. Think of it as an origami. You fold the paper horizontally at the center, so postive and negative parts are now packed within the upper half of the paper.
>
> (2) Cut the graph along the dotted lines and fold them back. But now they will overlap. To differentiate between high and low numeric values, color saturation is used: darkest blue represents largest positive numbers; medium blue stands for moderate values; and pale blue represents the smallest positive values. Similarly on the negative side. Note that there are three shades each of blue and red, corresponding to the three levels of cut we have made.

Is this what you're after? I must say I don't find them super intuitive but that may be just me.

**Comment by cassiobatista** [comment]
@poplarShift yes, that's exactly what I'm after. I'm new to data viz but I find myself in the very same situation stated by the link you provided:
> Line graphs are best suited for Time Series data. But if a large number of such plots have to be compactly presented on a cramped dashboard, they tend to become less useful. [...] Horizon plots are ideally suited for this situation.

**Comment by poplarShift** [comment]
Well with bokeh, panel, and other interactive technology around, the cramped dashboard argument does not really convince me :) though in print it may be worth the hassle of explaining the reader what all of the colours mean.

As a first solution (workaround?) one could implement the wrapping manually, transforming one timeseries into one timeseries per section of y coordinate,  and then overlay them as `Area` elements. Could even be done as `operation`.

**Comment by cassiobatista** [comment]
I'm working (trying to work, at least) on workaround based on Bokeh's [ridge plot](https://bokeh.pydata.org/en/latest/docs/gallery/ridgeplot.html), but I'm having some [interaction issues](https://discourse.bokeh.org/t/tools-in-ridge-plot-patches-hover-and-range/3709) even with `HoverTool` which I believe has been highly used and supported.

Furthermore, as you didn't find this implementation request reasonably useful, do you have any other option in mind to visualise multiple time-series data stream in a single plot?

**Comment by poplarShift** [comment]
Well actually now that I know what it is it doesn't sound that complicated anymore. Your example above though could have profited from a colorbar.

In the end one can see those charts as a twist on normal heatmaps where some of the small-scale gradients are not shown by color but by height inside each row, if that makes any sense.

As I said, you can just do the clipping and folding yourself, then make an NdOverlay of several hv.Area elements and maybe get some inspiration here: https://github.com/poplarShift/pyviz-recipes/blob/master/notebooks/styling_by_attribute.ipynb on styling their colors according to a colormap. I currently don't have a use case for it but if anyone wants to implement it more principally as an hv.Operation that shouldn't be too hard.
